index.bs

```diff
@@ -114,6 +122,154 @@ let sanitizedFragment = s.sanitize(userControlledInput);
+
+If the user data is available in string form and we wish to
+directly insert the sanitized subtree into the DOM, we can do so as follows:
+
+<div class="example">
+```js
+const user_string = "...";  // The user string.
+const sanitizer = new Sanitizer( ... );  // Our Sanitizer;
+
+// We want to insert the HTML in user_string into a target element with id
+// target. That is, we want the equivalent of target.innerHTML = value, except
+// without the XSS risks.
+document.getElementById("target").SetHTML(user_string, sanitizer);
+```
+</div>
+
+### Case 3: Sanitizing a String with a Given Context. ### {#string-context-case-3}
+
+If the user data is available in string form and we wish to sanitize it now
+but apply the result to the DOM later, then we can do so, but we must inform
+the Sanitizer about the context that it will be used in. To prevent context
+confusion we will return the result in a container that contains both the
+result, and also the parse context. Conveniently, this container already
+exists, and it is the node itself!
+
+<div class="example">
+```js
+// We would like to re-use a certain piece of user input repeatedly, to insert
+// it in multiple elements on the page. All these elements will be <div>
+// elements.
+const user_string = "...";  // The user string.
+const sanitizer = new Sanitizer( ... );  // Our Sanitizer.
+
+const sanitized = sanitizer.sanitizeFor("div", user_string);
+sanitized instanceof HTMLDivElement  // true. The Sanitizer has given us a node.
+
+// ... later, in the same program ...
+for (let elem = ... of ...) {
+  // We expect all of our "elem" instances to be of the same type we have
+  // used in the .sanitizeFor call above. If we had an assertion library, we
+  // could check this as follows:
+  assert_true(elem instanceof sanitized.constructor);  // Assuming assert_true, like in WPT tests.
+  elem.replaceChildren(sanitized.children);
+}
+
+// Instead of:
+elem.replaceChildren(sanitized.children);
+// we might also write:
+elem.innerHTML = sanitized.innerHTML;
+// This should have the same effect, except be slower, since this will trigger
+// un-parsing and then re-parsing the node tree which we already have
+// available as a node tree. So we recommend to stick with the former version.
+```
+</div>
+
+### The Other Case ### {#string-context-case-other}
+
+What if neither of these cases works with a given application structure, and a
+string-to-string operation is required? In this case, the developer is free to
+take the sanitization result and remove it from its context. In this case, the
+responsibility to prevent mXSS-class attacks that stem from mis-applying those
+strings in an inappropriate context remains with the developer.
+
+<div class="example">
+```js
+const user_string = "...";  // The user string.
+const sanitizer = new Sanitizer( ... );  // Our Sanitizer.
+
+// We plan to insert this string into a <div> container, but for $REASONS we
```

Comment:
I think it's useful to have this use-case covered even though we discourage it.
To further emphasize that this is a discouraged use-case, let's rewrite this from a "We" to a "The developer" point-of-view, OK?

E.g.,

```
// The developer plans to insert this string into a <div> element, but has to
// keep this around as a string (instead of an element). It's important that the
// developer remembers the parsing context and MUST NOT use this in
// a different context in order to prevent mXSS attacks.
```

or something along those lines.

Reply:
Done.